src/core/mac/mac.cpp

```diff
@@ -1091,6 +1091,10 @@ void Mac::ReceiveDoneTask(Frame *aFrame, ThreadError aError)
         break;
     }
 
+    // Duplicate Address Protection
+    VerifyOrExit(memcmp(&srcaddr.mExtAddress, &mExtAddress, sizeof(srcaddr.mExtAddress)) != 0,
```

Comment:
This is such a fundamental check, we should probably move this up to just after `GetSrcAddr()` above.